### DISCUSSION TITLE
Does remix-auth work with remix-utils’ typedSessionStorage?

### DISCUSSION THREAD

**Discussion by brandonpittman** [sergiodxa/remix-auth#214] [Q&A] [answered]
Using a `typedSessionStorage`, I can't call `authenticator.isAuthenticated(request)` (`undefined` property error), but I can call `authenticator.isAuthenticated(session)`.

**Comment by sergiodxa** [comment, marked as answer]
I use them together here https://github.com/sergiodxa/sergiodxa.com/blob/ca1fa0d9c7f8d92322b01f009951bbfdd54c6214/services/auth.ts#L53-L60 and I'm calling isAuthenticated with the request object https://github.com/sergiodxa/sergiodxa.com/blob/ca1fa0d9c7f8d92322b01f009951bbfdd54c6214/app/root.tsx#L59-L61.

Ensure that your schema marks everything as optional https://github.com/sergiodxa/sergiodxa.com/blob/ca1fa0d9c7f8d92322b01f009951bbfdd54c6214/services/auth.ts#L20-L25.

**Reply by brandonpittman** [reply]
I’ll double check this later. Thanks.

**Reply by brandonpittman** [reply]
Seems like it was because I was creating the authenticator and session storage in different files and the session storage wasn't being created before the authenticator was instantiated.
